### PR TITLE
Updating with issues

### PR DESCRIPTION
**Pull request by Emilio-Sakr**
Pull request for #1 #3 #4 #6 #9

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+secret=deff1952d59f883ece260e8683fed21ab0ad9a53323eca4fasdewr344sd
```
Comment (AnthonyDTrad) [inline .env:1]: Its a really bad practice to store sensitive config as a plain text. 
See credentials and keyvaults to know more on how to store secrets.

File: WeatherPredictorApi/SQLApp/crud.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+from xmlrpc.client import Boolean
```
Comment (AnthonyDTrad) [inline WeatherPredictorApi/SQLApp/crud.py:1]: This file is generally called dbContext. The context is the file where you manipulate and work with queries

File: WeatherPredictorApi/SQLApp/crud.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def getUserById(db: Session, id : int):
+    return db.query(models.user).filter(models.user.id == id).first()
+
+
+def getAllUsers(db: Session, skip: int = 0, limit: int = 100):
+    return db.query(models.user).offset(skip).limit(limit).all()
+
+def hashpassword(user, salt):
```
Comment (AnthonyDTrad) [inline WeatherPredictorApi/SQLApp/crud.py:26]: Why the class that manipulates the database access should know how to hash a password? is it the db's responsibility?

File: WeatherPredictorApi/SQLApp/crud.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    return bcrypt.hashpw(user.password.encode('utf-8'), salt)
+
+def dehashpassword(user, data) -> Boolean:
```
Comment (AnthonyDTrad) [inline WeatherPredictorApi/SQLApp/crud.py:29]: Same here

File: WeatherPredictorApi/SQLApp/crud.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    return dbUser
+
+# def deleteUser(db: Session, email: str):
+#     u = db.query(models.user).filter(models.user.email == email).delete()
+#     db.commit()
+#     return u
+
+
+
+def getCity(db: Session, name : str):
```
Comment (AnthonyDTrad) [inline WeatherPredictorApi/SQLApp/crud.py:48]: The goal is to code something that scales. Let's say we add 100 entities after 1 month. We will end up with a huge unmanageable file

File: WeatherPredictorApi/SQLApp/models.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+class user(Base):
+    __tablename__='users'
+    
+    id = Column(Integer, primary_key=True)
+    username = Column(String)
+    email = Column(String)
+    password = Column(String)
+    salt = Column(String)
+
+class city(Base):
```
Comment (AnthonyDTrad) [inline WeatherPredictorApi/SQLApp/models.py:14]: Scalable? When we add 100 models. How many lines of code we will have?

File: WeatherPredictorApi/__init__.py [hunk 7]
```diff
@@ -1,14 +1,39 @@
...
+from fastapi import FastAPI, Depends
+from WeatherPredictorApi.auth.authbearer import JWTBearer
 from WeatherPredictorApi.routers import cities
+from WeatherPredictorApi.routers.internal import internalcities
+from WeatherPredictorApi.auth import userauth
+from WeatherPredictorApi.routers import ws
+from WeatherPredictorApi.SQLApp.database import Base, engine
 
-
+Base.metadata.create_all(bind=engine)
```
Comment (AnthonyDTrad) [inline WeatherPredictorApi/__init__.py:9]: So the API should know that he should drope and recreate the db ?

File: WeatherPredictorApi/routers/cities.py [hunk 8]
```diff
@@ -1,14 +1,48 @@
...
+    return getAllCities(db)
+
+@Cities.get('/location/{city}')
+async def getLocation(city: str, db: Session = Depends(get_db)):
+    city = getCity(db, city)
+    if city: return city
+    else: raise HTTPException(status_code=404, detail='city does not exist')
+
+@Cities.get('/weather/{city}')
+async def getWeather(city: str, db: Session = Depends(get_db)) -> Dict:
```
Comment (AnthonyDTrad) [inline WeatherPredictorApi/routers/cities.py:23]: Will need to ask youo about this

**Review by AnthonyDTrad — CHANGES_REQUESTED** [review]
Got a first look, can you check those comments?

**Comment by Emilio-Sakr** [comment]
Thank you for your remarks, 
to fix them i'd like to create another branch (updating-with-issues-v2) because I want to restructure the project in better way. i'll disregard this pull request and create another.

**Pull request closed without merging**